### PR TITLE
Epic/cv2-3939: replace table view with cards

### PR DESCRIPTION
**Pull request by danielevalverde**
## Description

- Delete the GraphQL field list_columns from TeamType, delete the respective model method and any other dependent code and related tests
- Expose  projectMedia.fact_check_published_on field
References: CV2-3939

## Things to pay attention to during code review

removed all the code related to  list_columns

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

**Comment by danielevalverde** [comment]
> We just merged this, no? :D
> 
> [8188030](https://github.com/meedan/check-api/commit/81880302c2ee84e68195c55892f15082224e54d0)

Thanks, @caiosba  :)
I updated with develop now

**Comment by caiosba** [comment]
So there is nothing left here - can we close, then?

**Comment by amoedoamorim** [comment]
> So there is nothing left here - can we close, then?

There's Dani's PR agains the epic branch, then we'll merge the epic branch, but that's just leading to double reviewing the same stuff since I've already merged the other changes :shrug:

**Comment by caiosba** [comment]
I just clarified with @danielevalverde - yes, this is the epic branch, let's keep it.

**Comment by brianfleming** [comment]
this can/should be merged now too right @danielevalverde and @amoedoamorim ?

**Comment by brianfleming** [comment]
> this can/should be merged now too right @danielevalverde and @amoedoamorim ?

any reply to my question @danielevalverde or @amoedoamorim ? Asking because I notice the graphql warnings left in the output about the old pages

File: app/graph/types/team_type.rb [hunk 1]
```diff
@@ -39,7 +39,7 @@ class TeamType < DefaultObject
   field :get_language_detection, GraphQL::Types::Boolean, null: true
   field :get_report, JsonStringType, null: true
   field :get_fieldsets, JsonStringType, null: true
-  field :list_columns, JsonStringType, null: true
+  field :get_data_report_url, GraphQL::Types::String, null: true
```
Comment (caiosba) [inline app/graph/types/team_type.rb:42]: This is duplicating line 115

File: db/seeds.rb
Comment (caiosba) [inline db/seeds.rb:1]: These changes seem unrelated... maybe you need to rebase with `develop`?

**Review by caiosba — CHANGES_REQUESTED** [review]
Dani, please check my comments but also:

1) Be sure it's up-to-date with latest `develop`
2) Run integration tests against this branch

File: app/models/concerns/team_duplication.rb [hunk 2]
```diff
@@ -42,8 +42,6 @@ def self.duplicate(t, custom_slug = nil, custom_name = nil)
 
     def self.modify_settings(old_team, new_team)
       team_task_map = self.team_task_map
```
Comment (caiosba) [inline app/models/concerns/team_duplication.rb:44]: @danielevalverde you can delete this line too

**Review by caiosba — CHANGES_REQUESTED** [review]

**Review by caiosba — CHANGES_REQUESTED** [review]
@danielevalverde I have good news and bad news.

Bad news: I didn't pay attention to the code coverage report before, so there is still a bit more work to be done here.

Good news: We have more lines of code to be deleted! :tada: 

According to the [coverage report](https://app.travis-ci.com/github/meedan/check-api/jobs/623712490#L4515), we can delete:

```
app/models/concerns/project_media_getters.rb: 76
lib/check_search.rb: 529, 531, 536
```

- This method, which is not used anywhere anymore: https://github.com/meedan/check-api/blob/epic/cv2-3939-replace-table-view-with-cards/app/models/concerns/project_media_getters.rb#L75-L77
- This whole block, since we don't sort by tasks on the frontend side anymore: https://github.com/meedan/check-api/blob/epic/cv2-3939-replace-table-view-with-cards/lib/check_search.rb#L526-L552

File: lib/check_search.rb [hunk 3]
```diff
@@ -523,34 +523,7 @@ def team_tasks_conditions
...
-                    { term: { 'task_responses.team_task_id': team_task_id } },
-                    { exists: { field: 'task_responses.value' } }
-                  ]
-                }
-              }
-            }
-          }
-        }
-      ]
-    elsif SORT_MAPPING.keys.include?(@options['sort'].to_s)
```
Comment (caiosba) [inline lib/check_search.rb:553]: Please keep this line but change the `elsif` by `if`.

**Review by caiosba — CHANGES_REQUESTED** [review]

**Review by caiosba — APPROVED** [review]
Thanks Dani!

**Pull request merged**